dlup/cli/mask.py

```diff
@@ -0,0 +1,197 @@
+    """Perform the mask conversion to polygon."""
+    mask_filename = args.MASK_FN
+    output_filename = args.OUTPUT_FN
+    tile_size = cast(tuple[int, int], (args.tile_size,) * 2)
+    tile_overlap = cast(tuple[int, int], (args.tile_overlap,) * 2)
+
+    # Prepare output directory.
+    output_filename.parent.mkdir(parents=True, exist_ok=True)
+
+    dataset = TiledROIsSlideImageDataset.from_standard_tiling(
+        mask_filename,
+        tile_size=tile_size,
+        tile_overlap=tile_overlap,
+        mpp=None,
+        tile_mode=TilingMode.overflow,
+        crop=False,
+        backend=ImageBackend.TIFFFILE,
+        interpolator=Resampling.NEAREST,
+    )
+    target_mpp = args.mpp
+    if target_mpp is None:
+        scaling = 1.0
+    else:
+        scaling = dataset.slide_image.get_scaling(target_mpp=target_mpp)
+
+    # Parse the labels
+    if args.labels is None:
+        index_map = {1: "label"}
+    else:
+        index_map = {}
+        for pair in args.labels.split(","):
+            name, index = pair.split("=")
+            if not index.isnumeric():
+                raise argparse.ArgumentTypeError(f"Expected a key-pair of the form 1=tumor,2=stroma")
+            index = float(index)
+            if not index.is_integer():
+                raise argparse.ArgumentTypeError(f"Expected a key-pair of the form 1=tumor,2=stroma")
+            index = int(index)
+            if index == 0:
+                raise argparse.ArgumentTypeError(f"0 is not a proper index. Needs to be at least 1.")
+            index_map[index] = name.strip()
+
+    polygons = dataset_to_polygon(dataset, index_map=index_map, num_workers=args.num_workers, scaling=scaling)
+    wsi_annotations = [
+        WsiSingleLabelAnnotation(
+            label=label,
+            type=AnnotationType.POLYGON,
+            coordinates=[Polygon(polygons[label], label=label)],
+        )
+        for label in polygons
+    ]
+
+    slide_annotations = WsiAnnotations(wsi_annotations)
+
+    if not args.separate:
+        with open(output_filename, "w") as f:
+            json.dump(slide_annotations.as_geojson(split_per_label=False), f, indent=2)
+    else:
+        jsons = slide_annotations.as_geojson(split_per_label=True)
+        for label, json_dict in jsons:
+            suffix = output_filename.suffix
+            name = output_filename.with_suffix("").name
+            new_name = name + "-" + label
+            new_filename = (output_filename.parent / new_name).with_suffix(suffix)
+            with open(new_filename, "w") as f:
+                json.dump(json_dict, f, indent=2)
+
+
+def register_parser(parser: argparse._SubParsersAction):
+    """Register mask commands to a root parser."""
+    wsi_parser = parser.add_parser("mask", help="WSI mask parser")
+    wsi_subparsers = wsi_parser.add_subparsers(help="WSI mask subparser")
+    wsi_subparsers.required = True
+    wsi_subparsers.dest = "subcommand"
+
+    # Tile a slide and save the tiles in an output folder.
```

Comment:
This appears to be a comment from a copy-paste from the `wsi tiling` argparser.

Please delete